src/Keboola/SSHTunnel/SSH.php

```diff
@@ -31,42 +27,60 @@ public function generateKeyPair()
     }
 
     /**
-     *
-     * $user, $sshHost, $localPort, $remoteHost, $remotePort, $privateKey, $sshPort = '22'
+     * Open SSH tunnel defined by config
      *
      * @param array $config
-     *  - user
-     *  - sshHost
-     *  - sshPort
-     *  - localPort
-     *  - remoteHost
-     *  - remotePort
-     *  - privateKey
+     *
+     * Configuration fields:
+     *
+     * - user: (string) SSH proxy username. Required.
+     * - sshHost: (string) SSH proxy hostname. Required.
+     * - sshPort: (string) SSH protocol port. Optional, default 22.
+     * - localPort: (string) local port. Optional, default 33006.
+     * - remoteHost: (string) destination machine hostname. Required.
+     * - remotePort: (string) destination machine port. Required.
+     * - privateKey: (string) SSH private key. Required.
+     * - compression: (bool) whether to use compression. Optional, default false.
+     *
      *
      * @return Process
      * @throws SSHException
      */
-    public function openTunnel(array $config)
+    public function openTunnel(array $config): Process
     {
-        $missingParams = array_diff(
-            ['user', 'sshHost', 'sshPort', 'localPort', 'remoteHost', 'remotePort', 'privateKey'],
-            array_keys($config)
-        );
+        $process = new Process($this->createSshCommand($config));
+        $process->setTimeout(60);
+        $process->start();
 
-        if (!empty($missingParams)) {
-            throw new SSHException(sprintf("Missing parameters '%s'", implode(',', $missingParams)));
+        while ($process->isRunning()) {
+            sleep(1);
```

Comment:
nechceš tam dát usleep třeba na 50ms - vteřina mi přijde zbytečné prodlužování...

Reply:
Celkově vlastně ale nevím, proč to máš takhle. Nestačilo by try/catch okolo $process->mustRun()? Stejně ten procves nevrátíš dřív než je hotový...

Reply:
JJ, tiez myslim, ze by tam mohlo byt mustRun()